src/app/shared/image/image-upload.component.ts

```diff
@@ -126,6 +117,24 @@ export class ImageUploadComponent implements DoCheck {
     this.reader.readAsDataURL(file);
   }
 
+  fileDimensionsAcceptable(width: number, height: number): boolean {
+    if (!this.suggestSize) {
+      if (width < this.minWidth || height < this.minHeight) {
+        this.imgError = `The image provided is only ${width} x ${height} px
+                         but should be at least ${this.minWidth} x ${this.minHeight} px.`;
+        return false;
+      } else if ((this.maxWidth && width > this.maxWidth) || (this.maxHeight && height > this.maxHeight)) {
+        this.imgError = `The image provided is ${width} x ${height} px
+                         but should not exceed ${this.maxWidth} x ${this.maxHeight} px.`;
+        return false;
+      } else if (this.square && width !== height) {
```

Comment:
I wondered if we wanted to enforce square without enforcing min or max, but it looks like we're not doing that, so 👍